From 6508bc8a3440525384fcfcd8ad55a4cd5c79b912 Mon Sep 17 00:00:00 2001
From: Kenton Groombridge <me@concord.sh>
Date: Thu, 19 May 2022 11:43:44 -0400
Subject: [PATCH] systemd, udev: allow udev to read systemd-networkd runtime

udev searches for .link files and applies custom udev rules to devices
as they come up.

Upstream-Status: Backport
[https://github.com/SELinuxProject/refpolicy/commit/998ef975f38c70d57e7220b88ae5e62c88ebb770]

Thanks-To: Zhao Yi
Signed-off-by: Kenton Groombridge <me@concord.sh>
Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/udev.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/udev.te b/policy/modules/system/udev.te
index 4c5a690fb..8e243c0f2 100644
--- a/policy/modules/system/udev.te
+++ b/policy/modules/system/udev.te
@@ -270,6 +270,8 @@ ifdef(`init_systemd',`
 	systemd_read_hwdb(udev_t)
 	systemd_read_logind_sessions_files(udev_t)
 	systemd_read_logind_runtime_files(udev_t)
+	# udev searches for .link files and applies custom udev rules
+	systemd_read_networkd_runtime(udev_t)
 
 	optional_policy(`
 		init_dbus_chat(udev_t)
-- 
2.25.1

